### PR TITLE
feat: add HoverifyOptions.scrollBoundaries

### PR DESCRIPTION
**Pull request by lguychard**
Rel  [RFC 76](https://docs.google.com/document/d/1albi6HaAU9JAqdQXjsE-50NidxG4icE-36XqxxKgfM4/edit)

Introduces `HoverifyOptions.scrollBoundaries`, an array of `HTMLElement`, which would typically contain elements with a lower z-index than the hover overlay but a higher z-index than the code view, such as a sticky file header.

At runtime, in reaction to scroll events, hide the hover overlay when it overlaps with a hover boundary.

`IntersectionObserver` could not be used here, as it requires an ancestor relationship between the root and target elements.

**Before**:

<details>

</details>

**After**:

<details>

</details>

### CODE REVIEW

File: src/hoverifier.ts [hunk 1]
```diff
@@ -897,6 +922,29 @@ export function createHoverifier<C extends object, D, A>({
...
+                                return scrollEvents.pipe(
+                                    filter(() => scrollBoundaries.some(elementOverlaps(hoveredTokenElement)))
+                                )
+                            })
+                        )
+                        .subscribe(() => {
+                            resetHover()
+                        })
+                )
+            }
```
Comment (felixfbecker) [inline src/hoverifier.ts]: One thought I had was whether this should be in the `hoverify` method vs in the `createHoverifier` function above where all the other logic is. It does seem a little weird that we subscribe to the final Observable of state, which will emit a new hover state, but then in some cases we immediately hide the hover (which is another state emission that undoes the previous).

File: src/helpers.ts [hunk 2]
```diff
@@ -31,3 +31,17 @@ export const scrollIntoCenterIfNeeded = (container: HTMLElement, content: HTMLEl
         container.scrollTop = scrollTop
     }
 }
+
+/**
+ * Returns a curried function that retusn `true` if `e1` and `e2` overlap.
+ */
+export const elementOverlaps = (e1: HTMLElement) => (e2: HTMLElement) => {
```
Comment (felixfbecker) [inline src/helpers.ts]:
```suggestion
export const elementOverlaps = (e1: HTMLElement) => (e2: HTMLElement): boolean => {
```

**Comment by felixfbecker** [comment]
Could this be unit tested in Karma?

File: src/helpers.ts [hunk 3]
```diff
@@ -31,3 +31,17 @@ export const scrollIntoCenterIfNeeded = (container: HTMLElement, content: HTMLEl
         container.scrollTop = scrollTop
     }
 }
+
+/**
+ * Returns a curried function that retusn `true` if `e1` and `e2` overlap.
```
Comment (lguychard) [inline src/helpers.ts]:
```suggestion
 * Returns a curried function that returns `true` if `e1` and `e2` overlap.
```

File: src/hoverifier.ts (hunk 1 above)
Comment (lguychard) [reply src/hoverifier.ts]: @felixfbecker good point, I did this in c753c88. I think this can be tested in Karma, I'll work on that next.

**Pull request merged**